lib/ex_format/lines.ex

```diff
@@ -0,0 +1,15 @@
+defmodule ExFormat.Lines do
+  def initialize_lines_store(string) do
+  	lines = String.split(string, "\n")
+    Agent.start_link(fn -> %{} end, name: :lines)
+    for {line, i} <- Enum.with_index(lines) do
+      update_line(i + 1, String.trim(line))
+    end
```

Comment:
This will do a lot of work doing all the calls to the agent. Every time you update a line you will talk with the agent process. Also by convention when starting an agent you expose `start_link` so I would do

```elixir
def start_link(code_string) do
  lines =
    code_string
    |> String.split("\n")
    |> Enum.map(&String.trim/1)

  Agent.start_link(fn -> lines end, name: :lines)
end
```

Then I believe we can even ditch `update_line/2`.